### PR TITLE
xtensa: sample_controller32: for Xtensa MPU support

### PR DESCRIPTION
**Pull request by dcpleung**
The overlay comes from:

  https://github.com/jcmvbkbc/xtensa-toolchain-build
    commit 435c5e8b108de520565886f3f2349cc6be3fe712
    under directory overlays/

This would allow us to use QEMU to test the Xtensa MPU code on main Zephyr repo.

### CODE REVIEW

**Review by stephanosio — COMMENTED** [review]
Need to update `ci.yml` to build the new `xtensa-sample_controller32_zephyr-elf` toolchain.

Also QEMU changes from https://github.com/zephyrproject-rtos/qemu/pull/8 should be pulled as well.

**Comment by dcpleung** [comment]
I have changed `ci.yml` and the QEMU SHA. Hopefully I have modified in the correct place.

**Comment by dcpleung** [comment]
This needs https://github.com/zephyrproject-rtos/qemu/pull/9 to fix QEMU build.

**Comment by stephanosio** [comment]
@dcpleung https://github.com/zephyrproject-rtos/qemu/pull/9 is merged. Please update this PR to point to the merged commit.

**Comment by dcpleung** [comment]
> @dcpleung [zephyrproject-rtos/qemu#9](https://github.com/zephyrproject-rtos/qemu/pull/9) is merged. Please update this PR to point to the merged commit.

Done. Updated the QEMU `SRCREV`.

**Comment by stephanosio** [comment]
Test run: https://github.com/zephyrproject-rtos/sdk-ng/actions/runs/11062364153/job/30736729508

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1582,6 +1586,9 @@ jobs:
             xtensa-sample_controller_zephyr-elf)
               PLATFORM_ARGS+="-p qemu_xtensa "
               ;;
+            xtensa-sample_controller32_zephyr-elf)
+              PLATFORM_ARGS+="-p qemu_xtensa/sample_controller32/mpu "
+              ;;
```
Comment (stephanosio) [inline .github/workflows/ci.yml:1591]: This needs to be disabled until the board support is merged on the Zephyr side; otherwise, CI will fail.

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stephanosio) [reply .github/workflows/ci.yml]: Done

**Review by stephanosio — APPROVED** [review]

**Pull request merged**